executor/src/witgen/jit/interpreter.rs

```diff
@@ -208,63 +252,94 @@ impl<T: FieldElement> EffectsInterpreter<T> {
+                            id: c.id,
+                            ptype: PolynomialType::Constant,
+                        };
+                        vars[*idx] = fixed_data.fixed_cols[&poly_id].values_max_size()
+                            [usize::try_from(row_offset + c.row_offset as i64).unwrap()];
+                    }
+                    InterpreterAction::ReadParam(idx, i) => {
+                        vars[*idx] = get_param(params, *i);
+                    }
+                    InterpreterAction::WriteCell(idx, c) => {
+                        data.data.set(
+                            (row_offset + c.row_offset as i64).try_into().unwrap(),
+                            c.id,
+                            vars[*idx],
+                        );
+                    }
+                    InterpreterAction::WriteParam(idx, i) => {
+                        set_param(params, *i, vars[*idx]);
+                    }
+                    InterpreterAction::MachineCall(id, arguments) => {
+                        // we know it's safe to escape the references here, but the compiler doesn't, so we use unsafe
+                        let mut args = arguments
+                            .iter()
+                            .map(|a| match a {
+                                MachineCallArgumentIdx::Unknown(idx) => {
+                                    let var = &mut vars[*idx] as *mut T;
+                                    LookupCell::Output(unsafe { var.as_mut().unwrap() })
+                                }
+                                MachineCallArgumentIdx::Known(idx) => {
+                                    let var = &vars[*idx] as *const T;
+                                    LookupCell::Input(unsafe { var.as_ref().unwrap() })
+                                }
+                            })
+                            .collect::<Vec<_>>();
+                        mutable_state.call_direct(*id, &mut args[..]).unwrap();
+                    }
+                    InterpreterAction::Assertion(e1, e2, expected_equal) => {
+                        let lhs_value = e1.evaluate(&mut eval_stack, &vars);
+                        let rhs_value = e2.evaluate(&mut eval_stack, &vars);
+                        if *expected_equal {
+                            assert_eq!(lhs_value, rhs_value, "Assertion failed");
+                        } else {
+                            assert_ne!(lhs_value, rhs_value, "Assertion failed");
+                        }
+                    }
+                    InterpreterAction::Branch(test, if_branch, else_branch) => {
+                        // push the current block on the stack to continue execution once the branch is done
+                        block_stack.push(iter);
+                        // test the condition
+                        let condition = match test {
+                            BranchTest::Equal { var, value } => vars[*var] == *value,
```

Comment:
This could be a function of BranchTest. That way, we could encapsulate the logic of what the enum alternatives mean inside BrachTest and only provide functions `var()` and `test(value_to_test)`